sdk/core/azure-core-test/src/main/java/com/azure/core/test/utils/TestProxyDownloader.java

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.test.utils;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.logging.LogLevel;
+import org.apache.commons.compress.archivers.ArchiveEntry;
+import org.apache.commons.compress.archivers.ArchiveInputStream;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.archivers.zip.ZipArchiveInputStream;
+import org.apache.commons.compress.utils.IOUtils;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Locale;
+import java.util.zip.GZIPInputStream;
+
+/**
+ * Class for managing downloads of the test proxy
+ */
+public final class TestProxyDownloader {
+    private static final ClientLogger LOGGER = new ClientLogger(TestProxyDownloader.class);
+    private static final String TEST_PROXY_TAG = "test-proxy_1.0.0-dev.20230125.14";
```

Comment:
We should have a way to update this without requiring a code update as that means we'll need a release every time there is a fix to the test proxy.

Reply:
Is there a way to always use the latest tag? Is there a reason to always tie this to a specific tag?

Reply:
We don't want to be on latest as we want to be intentional about when we roll forward. There's a file in the repo (in the `eng`) directory that we can read that the engsys team updates when there is a new LKG. It's just not implemented here yet.